File: SCG Addon Taller/ControlesSBO/MatrixSBO.vb
Option Strict On
Option Explicit On

Imports System.Globalization
Imports SAPbouiCOM

Namespace ControlesSBO


    <CLSCompliant(False)> _
    Public Class MatrixSBO
        Inherits ControlSBO

        Protected _matrix As IMatrix

        Public Sub New(ByVal itemSBO As IItem, ByVal permitirBuscar As Boolean)
            MyBase.New(itemSBO, permitirBuscar)
        End Sub

        Protected Overrides Sub AsignaControlEspecifico()
            _matrix = DirectCast(ItemSBO.Specific, IMatrix)
        End Sub

        Public ReadOnly Property Matrix() As IMatrix
            Get
                Return _matrix
            End Get
        End Property

        Public Sub AsignaValorColumnaEditText(ByVal valor As String, ByVal nombreColumna As String, ByVal posicion As Integer)
            DirectCast(_matrix.Columns.Item(nombreColumna).Cells.Item(posicion).Specific, EditText).Value = valor
        End Sub

        Public Function ObtieneValorColumnaEditText(ByVal nombreColumna As String, ByVal posicion As Integer) As String
            Return DirectCast(_matrix.Columns.Item(nombreColumna).Cells.Item(posicion).Specific, EditText).Value
        End Function
    End Class
End Namespace

File: SCG.SBOFramework/UI/MatrixSBO.cs
using System.Collections.Generic;
using System.Globalization;
using SAPbouiCOM;

namespace SCG.SBOFramework.UI
{
    public abstract class MatrixSBO : ControlSBO<IMatrix>
    {
        public string TablaLigada { get; set; }

        protected MatrixSBO(IItem itemSBO)
            : base(itemSBO)
        {
        }

        protected MatrixSBO(string uniqueId) : base(uniqueId)
        {
        }

        protected MatrixSBO(string uniqueId, IForm formularioSBO) : base(uniqueId, formularioSBO)
        {
        }

        public void AsignaValorColumnaEditText(string valor, string nombreColumna, int posicion)
        {
            ((EditText) Especifico.Columns.Item(nombreColumna).Cells.Item(posicion).Specific).Value = valor;
        }

        public string ObtieneValorColumnaEditText(string nombreColumna, int posicion)
        {
            return ((EditText) Especifico.Columns.Item(nombreColumna).Cells.Item(posicion).Specific).Value;
        }

        public IMatrix Matrix 
        {
            get { return Especifico; }
        }

        public NumberFormatInfo NumberFormatInfo { get; set; }

        public abstract void LigaColumnas();
        public abstract void CreaColumnas();

    }
}